[JIT] Fix custom class method binding for const methods

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32555 [JIT] Serialize attributes and types in ClassType serialization
* #32553 [JIT] Support for registering C++ lambdas as methods on custom C++ class
* #32514 Revert "Remove __torch__ from custom class qualname"
* #32508 [JIT] Add torch.classes.load_library
* #32477 [JIT] Support returning tuple from custom bound C++ method
* **#32471 [JIT] Fix custom class method binding for const methods**
* #32470 [JIT] Test __getstate__ and __setstate__ for custom bound C++ classes
* #32260 [JIT] Passing custom class as arg

Differential Revision: [D19508249](https://our.internmc.facebook.com/intern/diff/D19508249)

Commit:
[JIT] Fix custom class method binding for const methods

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 2c7b0a25:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=2c7b0a2541a45faf3d0fe20e5ebccd7e9e955ade).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4312045) pytorch_linux_xenial_cuda10_1_cudnn7_py3_multigpu_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=89177493) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=89177493)) :snowflake:
<details>
<summary>
<code>Jan 22 00:49:34 AssertionError: 10 not less than or equal to 1e-05 :</code>
</summary>

```
Jan 22 00:49:34 ---------------------------------------------------------------------- 
Jan 22 00:49:34 Traceback (most recent call last): 
Jan 22 00:49:34   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 130, in wrapper 
Jan 22 00:49:34     self._join_processes(fn) 
Jan 22 00:49:34   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 211, in _join_processes 
Jan 22 00:49:34     self._check_return_codes(elapsed_time) 
Jan 22 00:49:34   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 231, in _check_return_codes 
Jan 22 00:49:34     self.assertEqual(p.exitcode, first_process.exitcode) 
Jan 22 00:49:34   File "/var/lib/jenkins/workspace/test/common_utils.py", line 890, in assertEqual 
Jan 22 00:49:34     super(TestCase, self).assertLessEqual(abs(x - y), prec, message) 
Jan 22 00:49:34 AssertionError: 10 not less than or equal to 1e-05 :  
Jan 22 00:49:34  
Jan 22 00:49:35 ---------------------------------------------------------------------- 
Jan 22 00:49:35 Ran 117 tests in 209.466s 
Jan 22 00:49:35  
Jan 22 00:49:35 FAILED (failures=1, skipped=4) 
Jan 22 00:49:35  
Jan 22 00:49:35 Generating XML reports... 
Jan 22 00:49:35 Traceback (most recent call last): 
Jan 22 00:49:35   File "test/run_test.py", line 457, in <module> 
Jan 22 00:49:35     main() 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "[JIT] Fix custom class method binding for const methods"


Differential Revision: [D19508249](https://our.internmc.facebook.com/intern/diff/D19508249)

[ghstack-poisoned]

dzhulgakov reviewed (approved):
It could have been done probably with some "decay" magic - but this works too :)

Commit:
Update on "[JIT] Fix custom class method binding for const methods"


Differential Revision: [D19508249](https://our.internmc.facebook.com/intern/diff/D19508249)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix custom class method binding for const methods"


Differential Revision: [D19508249](https://our.internmc.facebook.com/intern/diff/D19508249)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix custom class method binding for const methods"


Differential Revision: [D19508249](https://our.internmc.facebook.com/intern/diff/D19508249)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix custom class method binding for const methods"


Differential Revision: [D19508249](https://our.internmc.facebook.com/intern/diff/D19508249)

[ghstack-poisoned]
